CIAOS-Rust/src/main.rs

```diff
@@ -0,0 +1,291 @@
+use actix_web::{HttpServer, App, post, web, HttpResponse,Error };
+use futures::StreamExt;
+use bytes::BytesMut;
+
+use log::{info, error, warn};
+use bincode;
+use actix_web::error::ErrorInternalServerError;
+
+// storage.rs contains functionality related to writing files to storage and getting offset and size
+//as well as dealing with file retrieval from storage and deletion
+mod storage;
+use crate::storage::{write_files_to_storage, get_files_from_storage, delete_and_log};
+
+
+//keys and offset:size lists are managed here 
+mod database;
+use crate::database::{check_key,append_sql, update_file_db, upload_sql, get_offset_size_lists, delete_from_db, update_key_from_db};
+
+
+//offset and size of files are converted to binary to be stored in blob format in sql i.e deserialization and serialization fucntions 
```

Comment:
Add multi lines comments explaining in short sentences as to what it does 
/****
Comments

***/

Also why do we have to convert them to blob why can't I store them in string and JSON? Why is this approach better?

Reply:
yea my bad